contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -212,139 +202,89 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
 
         emit RevenueTokenClaimed(msg.sender, token, claimableToken);
-    }
-
-    /// @notice Allows ancestors to claim the royalty tokens and any accrued revenue tokens
-    /// @param ancestorIpId The ip id of the ancestor to whom the royalty tokens belong to
-    function collectRoyaltyTokens(address ancestorIpId) external nonReentrant whenNotPaused {
-        IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
-
-        address ipId = $.ipId;
-
-        if (DISPUTE_MODULE.isIpTagged(ipId)) revert Errors.IpRoyaltyVault__IpTagged();
-        if ($.isCollectedByAncestor[ancestorIpId]) revert Errors.IpRoyaltyVault__AlreadyClaimed();
 
-        // check if the address being claimed to is an ancestor
-        if (!_hasAncestorIp(ipId, ancestorIpId)) revert Errors.IpRoyaltyVault__ClaimerNotAnAncestor();
-
-        // transfer royalty tokens to the ancestor
-        uint32 ancestorsRoyalty = _getRoyalty(ipId, ancestorIpId);
-        IERC20Upgradeable(address(this)).safeTransfer(ancestorIpId, ancestorsRoyalty);
-
-        // save the amount of revenue tokens that are collectable by the ancestor
-        address[] memory tokenList = $.tokens.values();
-        uint256 unclaimedTokens = $.unclaimedRoyaltyTokens;
-        for (uint256 i = 0; i < tokenList.length; ++i) {
-            // the only case in which unclaimedRoyaltyTokens can be 0 is when the vault is empty and everyone claimed
-            // in which case the call will revert upstream with IpRoyaltyVault__AlreadyClaimed error
-            uint256 collectAmount = ($.ancestorsVaultAmount[tokenList[i]] * ancestorsRoyalty) / unclaimedTokens;
-            if (collectAmount == 0) continue;
-
-            $.collectableAmount[ancestorIpId][tokenList[i]] += collectAmount;
-        }
-
-        $.isCollectedByAncestor[ancestorIpId] = true;
-        $.unclaimedRoyaltyTokens -= ancestorsRoyalty;
-
-        emit RoyaltyTokensCollected(ancestorIpId, ancestorsRoyalty);
+        return claimableToken;
     }
 
-    /// @notice Collect the accrued tokens (if any)
-    /// @param ancestorIpId The ip id of the ancestor to whom the royalty tokens belong to
-    /// @param _tokens The list of revenue tokens to claim
-    function collectAccruedTokens(
-        address ancestorIpId,
-        address[] calldata _tokens
-    ) external nonReentrant whenNotPaused {
-        IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
-
-        if (DISPUTE_MODULE.isIpTagged($.ipId)) revert Errors.IpRoyaltyVault__IpTagged();
+    /// @notice Allows to claim revenue tokens on behalf of the ip royalty vault by token batch
+    /// @param snapshotId The snapshot id
+    /// @param tokenList The list of revenue tokens to claim
+    /// @param targetIpId The target ip id to claim revenue tokens from
+    function claimByTokenBatchAsSelf(
+        uint256 snapshotId,
+        address[] calldata tokenList,
+        address targetIpId
+    ) external whenNotPaused {
```

Comment:
Is the function for claim revenue token from another IP's vault to current vault? if so, could we find function name even more intuitive?

Reply:
The function is meant to claim to self - and self is the contract being called. Not only vaults claiming to themselves but also royalty policies claiming to themselves. Hence claimAsSelf.